### PR TITLE
Fix Mergify config

### PR DESCRIPTION
**Pull request by beatngu13**
@roesslerj pointed out that Mergify sometimes merges base into head, rather than rebasing head against base. This fix should solve this issue and also introduces less-verbose rule names.

@diba1013 we already talked about this a few weeks ago. Any other ideas? We should also keep an eye on this before adopting Mergify elsewhere. However, as soon as this is merged, I will update the other configs accordingly.

### CODE REVIEW

**Review by diba1013 — CHANGES_REQUESTED** [review]
Just to recap:

We want all branches up to date prior to merging. We have following use cases:
1. If non-release (into either master/release): rebase onto base, then merge.
2. If release (into master): merge base into head, then merge.

My question is: "Do we still need the rebase if the review is pending?". As this essentially is the cause for the rebase/merge conflict. 

If I recall our processes from the past, we just rebased to make it easier to merge by allowing the reviewer to simply press the merge button (as it will be always up to date) or to know if there are conflicts (which a good git diff will show anyways). We kept this for the rebazer so that it knows and can comment if there are conflicts. But since we always make sure that the branches are up to date just before merging, can we get rid of the rebase while the review is pending and just rebase once prior to merging?

Following use cases come to mind, if we rebase to keep the branch up to date:

1. rebase successful: No action needed.
2. review approved, rebase fails: author needs to manually do the rebase and resolve the conflict. The reviewer is unlikely to verify these changes afterwards (as we do not remove the approval if there are conflicts.
3. review pending, rebase fails: this is displayed in the checks (to my knowledge) and either the autor or the reviewer can simply request changes to resolve the conflicts.

If there are multiple PRs approved at the same time, one can hardly predict if there are conflicts.

@martin-v Am I missing something?

---

Either way, we could try to stop a merge if there are conflicts:

```yml
condition:
  - conflict=true
actions:
  comment:
    message: This PR needs some manual love. # Or similar
  dismiss_reviews:
    approved: true
    message: The author needs to resolve conflicts. Please re-review afterwards.  
```

File: .mergify.yml
Comment (beatngu13) [reply .mergify.yml:2]: Hmmm, I don't know. Even if Mergify doesn't handle this properly, Git/GitHub does. In the worst case, the same rebase happens twice I guess?

Anyways, what condition would you suggest?

**Comment by beatngu13** [comment]
>My question is: "Do we still need the rebase if the review is pending?".

Otherwise PRs without a review won't be up-to-date, only right before the merge. Then a review might happen on an outdated branch.

The rebazer also did rebase unreviewed PRs:

https://github.com/retest/rebazer/blob/develop/src/main/java/org/retest/rebazer/RebazerService.java#L68

File: .mergify.yml
Comment (diba1013) [reply .mergify.yml:2]: Maybe only rebase if review pending (i.e. invert the merge condition)? 
```
- "#approved-reviews-by<=0"
- "#changes-requested-reviews-by<=0"
```
I'm still unsure if we need the operator `=` or `<=`

**Review by diba1013 — APPROVED** [review]

**Comment by diba1013** [comment]
@beatngu13 You'll have to manually merge this.

**Pull request merged**
